csharp/rust/src/lib.rs

```diff
@@ -1,284 +1,472 @@
+
+    let ffi_handle = unsafe { Box::leak(Box::from_raw(in_client_ptr as *mut FFIHandle)) };
+    let handle = ffi_handle.handle.clone();
+    let routing_info = if in_routing_info.is_null() {
+        None
+    } else {
+        Some(unsafe {
+            match (*in_routing_info).to_redis() {
+                Ok(d) => d,
+                Err(e) => {
+                    logger_core::log(Error, 
+                        "csharp_ffi",
+                        format!(
+                            "Error while parsing route in string transformation: {:?}",
+                            e.to_string()
+                        ),
+                    );
+                    return match e {
+                        Utf8OrEmptyError::Utf8Error(e) => CommandResult::new_error(
+                            helpers::to_cstr_ptr_or_null(e.to_string().as_str()),
+                        ),
+                        Utf8OrEmptyError::Empty => {
+                            CommandResult::new_error(helpers::to_cstr_ptr_or_null(
+                                "Routing info incomplete, null value passed in string",
+                            ))
+                        }
+                    };
+                }
+            }
+        })
+    };
+    ffi_handle.runtime.spawn(async move {
+        logger_core::log(Trace, "csharp_ffi", "Entered command task with");
+        let data: redis::Value = match handle.command(cmd, args, routing_info).await {
+            Ok(d) => d,
+            Err(e) => {
+                logger_core::log(Error, 
+                    "csharp_ffi",
+                    format!(
+                        "Error handling command in task of csharp_command: {:?}",
+                        e.to_string()
+                    ),
+                );
+                let value = Value::simple_string_with_null(e.to_string().as_str());
+                match catch_unwind(|| unsafe {
+                    logger_core::log(Trace, 
+                        "csharp_ffi",
+                        "Calling command callback of csharp_command",
+                    );
+                    callback(callback_data as *mut c_void, false as c_int, value);
+                    logger_core::log(Trace, 
+                        "csharp_ffi",
+                        "Called command callback of csharp_command",
+                    );
+                }) {
+                    Err(e) => logger_core::log(Error, 
+                        "csharp_ffi",
+                        format!("Exception in C# callback: {:?}", e),
+                    ),
+                    _ => {}
+                };
+                return;
+            }
         };
+        unsafe {
+            let mut buffer = FFIBuffer::new();
 
-        let logger_level = logger_core::init(level.map(|level| level.into()), file_name_as_str);
-        logger_level.into()
-    }
+            // "Simulation" run
+            _ = Value::from_redis(&data, &mut buffer);
```

Comment:
What is purpose of that? debugging?

Reply:
See the documentation of `FFIBuffer`.  
Essentially, I "simulate" writing the data once to calculate the required byte size, then perform the actual write afterward. This enables a single memory allocation for all result cases (e.g., arrays containing strings), avoiding multiple allocations in favor of one contiguous block.  
In other words: a micro-optimization.